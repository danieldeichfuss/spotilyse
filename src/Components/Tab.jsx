import React from "react";
import mojs from "mo-js";

export default function Tab({ isActive, tabName, tabIndex, onClick }) {
  const shootLines = e => {
    const itemDim = e.currentTarget.getBoundingClientRect(),
      itemSize = {
        x: itemDim.right - itemDim.left,
        y: itemDim.bottom - itemDim.top
      },
      shapes = ["line", "zigzag"],
      colors = ["#2FB5F3", "#FF0A47", "#FF0AC2", "#47FF0A"];

    const chosenC = Math.floor(Math.random() * colors.length),
      chosenS = Math.floor(Math.random() * shapes.length);

    const burst = new mojs.Burst({
      left: 0,
      top: 0,
      radiusX: itemSize.x,
      radiusY: itemSize.y,
      count: 8,

      children: {
        shape: shapes[chosenS],
        radius: 10,
        scale: { 0: 1.5 },
        fill: "none",
        points: 7,
        stroke: colors[chosenC],
        strokeDasharray: "100%",
        strokeDashoffset: { "-100%": "100%" },
        duration: 350,
        easing: "quad.out",
        isShowEnd: false
      }
    });
    burst.tune({ x: e.pageX, y: e.pageY });
    burst.replay();
  };

  return (
    <li className="Tab" onClick={shootLines}>
      <a
        className={`Tab__link ${isActive ? "active" : ""}`}
        onClick={event => {
          event.preventDefault();
          onClick(tabIndex);
        }}
      >
        {tabName}
      </a>
    </li>
  );
}
