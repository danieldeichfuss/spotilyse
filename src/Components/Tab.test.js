import React from "react";
import ReactDOM from "react-dom";
import Tab from "./Tab";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Tab />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders page correctly", () => {
  const component = renderer.create(<Tab />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
