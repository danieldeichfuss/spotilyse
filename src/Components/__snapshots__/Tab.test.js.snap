// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page correctly 1`] = `
<li
  className="Tab"
  onClick={[Function]}
>
  <a
    className="Tab__link "
    onClick={[Function]}
  />
</li>
`;
